pyccel/codegen/printing/ccode.py

```diff
@@ -474,6 +476,24 @@ def get_declare_type(self, expr):
         else:
             return '{0} '.format(dtype)
 
+    def _print_FuncAddressDeclare(self, expr):
+        if len(expr.results) == 1:
+            ret_type = self.get_declare_type(expr.results[0])
+        elif len(expr.results) > 1:
+            msg = 'Multiple output arguments is not yet supported in c'
+            errors.report(msg+'\n'+PYCCEL_RESTRICTION_TODO, symbol=expr,
+                severity='fatal')
+        else:
+            ret_type = self._print(datatype('void')) + ' '
+        name = expr.name
+        if not expr.arguments:
+            arg_code = 'void'
+        else:
+            arg_code = ', '.join('{}'.format(self._print_FuncAddressDeclare(i))
+                        if isinstance(i, FunctionAddress) else '{0}{1}'.format(self.get_declare_type(i), i)
+                        for i in expr.arguments)
```

Comment:
Are you handling here a function as argument, which itself has a function argument?

Reply:
yes, except that it's not fully supported, I still don't properly extract information about argument types and return types of the most inner function argument

Reply:
Good. Can you please add a comment line (starting with `# TODO:`) with this explanation?